truss/util/user_config.py

```diff
@@ -0,0 +1,290 @@
+"""Tools to configure and track the Truss CLI state and behavior, e.g. auto-upgrades."""
+
+import datetime
+import logging
+import os
+import pathlib
+import shutil
+import subprocess
+from typing import Any, Optional, Union
+
+import packaging.version
+import pydantic
+import requests
+import tomlkit
+import tomlkit.container
+import tomlkit.items
+
+from truss.remote.remote_factory import load_config
+
+_SETTINGS_FiLE = "settings.toml"
+_STATE_FILE = "state.json"
+_PYPI_VERSION_URL = "https://pypi.org/pypi/truss/json"
+
+
+def _get_dir() -> pathlib.Path:
+    if "XDG_CONFIG_HOME" in os.environ:
+        base_dir = pathlib.Path(os.environ["XDG_CONFIG_HOME"])
+    else:
+        base_dir = pathlib.Path.home() / ".config"
+
+    settings_path = base_dir / "truss"
+    settings_path.mkdir(parents=True, exist_ok=True)
+    return settings_path
+
+
+def _truss_is_git_branch() -> bool:
+    if shutil.which("git") is None:
+        return False
+    source_path = pathlib.Path(__file__).resolve()
+    source_dir = source_path.parent
+    try:
+        result = subprocess.run(
+            "git rev-parse --is-inside-work-tree && git remote get-url origin",
+            cwd=source_dir,
+            shell=True,
+            text=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.DEVNULL,
+            check=True,
+        )
+        remote_url = result.stdout.strip().splitlines()[-1]
+        remote_matches = (
+            remote_url.rstrip("/") == "https://github.com/basetenlabs/truss"
```

Comment:
[low priority, non-blocking] nit, could we just look for includes "basetenlabs/truss", in case for people using ssh git authentication